run_sharded_tests.sh

```
@@ -0,0 +1,233 @@
+            shard3=$((batch+3))
+
+            if [ $shard1 -le $TOTAL_SHARDS ]; then
+                # Build message with only valid shards
+                msg="▶ Running Shard $shard1"
+                [ $shard2 -le $TOTAL_SHARDS ] && msg+=", $shard2"
+                [ $shard3 -le $TOTAL_SHARDS ] && msg+=", $shard3"
+                msg+=" in parallel..."
+                echo -e "${BLUE}$msg${NC}"
+
+                pids=()
+                local -a cmd1=()
```

Shellcheck (0.11.0)

[error] 'local' is only valid in functions.

(SC2168)